CHANGELOG.md

```diff
@@ -15,6 +15,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add dynamic setting allowing size > 0 requests to be cached in the request cache ([#16483](https://github.com/opensearch-project/OpenSearch/pull/16483))
 - Support installing plugin SNAPSHOTs with SNASPHOT distribution ([#16581](https://github.com/opensearch-project/OpenSearch/pull/16581))
 - Make IndexStoreListener a pluggable interface ([#16583](https://github.com/opensearch-project/OpenSearch/pull/16583))
+- Add create rule api logic for rule-based auto tagging ([#17335](https://github.com/opensearch-project/OpenSearch/pull/17335))
```

Comment:
We should add the project name in the beginning since it will help the viewer to set the context faster without reading the full line, e,g;
`[Rule based auto-tagging] Add create rule API `